tests/app/views/test_suppliers.py

```diff
@@ -1368,18 +1386,76 @@ def test_agreement_returned_at_timestamp_cannot_be_set(self):
             data = json.loads(response.get_data())
             assert_equal(data['frameworkInterest']['agreementReturnedAt'], '2012-12-12T00:00:00.000000Z')
 
-    def test_agreement_returned_at_is_unset_when_agreement_reutrned_flag_is_false(self):
-        self.supplier_framework_update(
-            0, 'digital-outcomes-and-specialists',
-            update={'agreementReturned': True})
+    def test_countersigned_at_timestamp_cannot_be_set(self):
+        with freeze_time('2012-12-12'):
+            response = self.supplier_framework_update(
+                0,
+                'digital-outcomes-and-specialists',
+                update={
+                    'agreementReturned': True,
+                    'countersigned': True,
+                    'countersignedAt': '2013-13-13T00:00:00.000000Z',
+                }
+            )
+            assert_equal(response.status_code, 200)
+            data = json.loads(response.get_data())
+            assert_equal(data['frameworkInterest']['countersignedAt'], '2012-12-12T00:00:00.000000Z')
+
+    def test_agreement_returned_at_is_unset_when_agreement_returned_flag_is_false(self):
```

Comment:
So we can do away with the unset tests, because wet shouldn't allow unsetting. But a test that setting one date leaves the value of the other date unchanged would be useful.